src/testing/factories/juju/Application.ts

```diff
@@ -1,33 +1,34 @@
 import type {
-  AdditionalProperties,
   ApplicationGetResults,
   Base,
   Value,
   ErrorResults,
 } from "@canonical/jujulib/dist/api/facades/application/ApplicationV15";
 import { Factory } from "fishery";
 
-export const applicationConfigFactory = Factory.define<AdditionalProperties>(
+export const applicationConfigFactory = Factory.define<
+  NonNullable<ApplicationGetResults["application-config"]>
+>(() => ({
+  trust: configFactory.build({
+    default: false,
+    description: "Does this application have access to trusted credentials",
+    source: "default",
```

Comment:
I'm not sure, but I think we could remove this line, as `source` will be inffered to be `"default"` from factory.

Reply:
Well spotted, I've fixed this too.